alternator/executor.cc

```diff
@@ -1789,7 +1789,13 @@ future<executor::request_return_type> executor::update_table(client_state& clien
         }
 
         co_await verify_permission(enforce_authorization, client_state_other_shard.get(), schema, auth::permission::ALTER);
-        auto m = co_await service::prepare_column_family_update_announcement(p.local(), schema, std::vector<view_ptr>(), group0_guard.write_timestamp());
+        std::vector<view_ptr> view_updates;
+        auto cf = p.local().data_dictionary().find_column_family(schema);
+        // Update all view so that they use the new base schema version.
+        for (auto&& view : cf.views()) {
+            view_updates.push_back(view_ptr(schema_builder(view).build()));
```

Comment:
I don't understand why announcing all the old pre-existing views is now necessary and wasn't necessary in the past. I have a vague memory that it was NOT necessary, because the theory is that view V1 doesn't care that view V2 was added to the schema. Why has it become important now?

Reply:
With this new approach we want to announce all views on base table update to create new versions of schemas for them with the new base info - we used to modify the existing schemas with new base info but after this series we're no longer able to modify them in place. We considered creating new view schemas with the same version but different base info but there's at least one place where that would be wrong - the schema registry holds only one variant of one view schema version, with one base info.
In the rebase this code is no longer here, but we're still doing this in `prepare_column_family_update_announcement` which is shared by all the cases handled in this commit.